src/main/java/org/opentripplanner/api/parameter/QualifiedModeSet.java

```diff
@@ -29,32 +33,103 @@ public QualifiedModeSet(String s) {
         }
     }
 
-    /**
-     * Modify an existing routing request, setting fields to reflect these qualified modes.
-     * This is intended as a temporary solution, and uses the current system of a single mode set,
-     * accompanied by some flags to help with routing.
-     */
-    public void applyToRoutingRequest(RoutingRequest req) {
+    public RequestModes getRequestModes() {
+        RequestModes requestModes = new RequestModes(
+            StreetMode.WALK,
+            StreetMode.WALK,
+            StreetMode.WALK,
+            Collections.emptySet()
+        );
 
-        if (qModes.isEmpty()) return;
+        if (qModes.isEmpty()) return requestModes;
 
-        /* Start with an empty mode set. */
-        TraverseModeSet modes = new TraverseModeSet();
-        req.setModes(modes);
-        
-        /* First, copy over all the unqualified modes and see if we are using transit. FIXME HACK */
+        // Set transit modes
         for (QualifiedMode qMode : qModes) {
-            modes.setMode(qMode.mode, true);
+             switch (qMode.mode) {
+                 case "RAIL":
+                     requestModes.transitModes.add(TransitMode.RAIL);
+                     break;
+                 case "SUBWAY":
+                     requestModes.transitModes.add(TransitMode.SUBWAY);
+                     break;
+                 case "BUS":
+                     requestModes.transitModes.add(TransitMode.BUS);
+                     break;
+                 case "TRAM":
+                     requestModes.transitModes.add(TransitMode.TRAM);
+                     break;
+                 case "FERRY":
+                     requestModes.transitModes.add(TransitMode.FERRY);
+                     break;
+                 case "AIRPLANE":
+                     requestModes.transitModes.add(TransitMode.AIRPLANE);
+                     break;
+                 case "CABLE_CAR":
+                     requestModes.transitModes.add(TransitMode.CABLE_CAR);
+                 case "GONDOLA":
+                     requestModes.transitModes.add(TransitMode.GONDOLA);
+                     break;
+                 case "FUNICULAR":
+                     requestModes.transitModes.add(TransitMode.FUNICULAR);
+                     break;
+             }
         }
-        boolean usingTransit = modes.isTransit();
-        
-        // We used to always set WALK to true, but this forced walking when someone wanted to use a bike.
-        // We also want it to be possible to force biking-only (e.g. this is done in some consistency tests).
-        // TODO clearly define mode semantics: does presence of mode mean it is allowable, preferred... ?
 
+        if (requestModes.transitModes.isEmpty()) {
+            requestModes.transitModes = new HashSet<>(Arrays.asList(TransitMode.values()));
```

Comment:
This adds all transit modes, when a street-only search is made

Reply:
I will change this. Feel free to change this mapping further, as we are not testing the REST API at Entur and in some cases we do not know the intended functionality.